Specs/Renderer/FramebufferManagerSpec.js

```diff
@@ -0,0 +1,446 @@
+        return;
+      }
+
+      var length = 2;
+      fbm = new FramebufferManager({
+        colorAttachmentsLength: length,
+      });
+      fbm.update(context, 1, 1);
+      for (var i = 0; i < length; ++i) {
+        var texture = fbm.getColorTexture(i);
+        expect(texture).toBeDefined();
+        expect(texture.width).toEqual(1);
+        expect(texture.height).toEqual(1);
+        expect(texture.pixelFormat).toEqual(PixelFormat.RGBA);
+        expect(texture.pixelDatatype).toEqual(PixelDatatype.UNSIGNED_BYTE);
+      }
+    });
+
+    it("creates depth attachments", function () {
+      fbm = new FramebufferManager({
+        color: false,
+        depth: true,
+        supportsDepthTexture: true,
+      });
+      fbm.update(context, 1, 1);
+      if (context.depthTexture) {
+        var texture = fbm.getDepthTexture();
+        expect(texture).toBeDefined();
+        expect(texture.width).toEqual(1);
+        expect(texture.height).toEqual(1);
+        expect(texture.pixelFormat).toEqual(PixelFormat.DEPTH_COMPONENT);
+        expect(texture.pixelDatatype).toEqual(PixelDatatype.UNSIGNED_INT);
+      } else {
+        var renderbuffer = fbm.getDepthRenderbuffer();
+        expect(renderbuffer).toBeDefined();
+        expect(renderbuffer.width).toEqual(1);
+        expect(renderbuffer.height).toEqual(1);
+        expect(renderbuffer.format).toEqual(
+          RenderbufferFormat.DEPTH_COMPONENT16
+        );
+      }
+    });
+
+    it("creates depth-stencil attachments", function () {
+      fbm = new FramebufferManager({
+        color: false,
+        depthStencil: true,
+        supportsDepthTexture: true,
+      });
+      fbm.update(context, 1, 1);
+      if (context.depthTexture) {
+        var texture = fbm.getDepthStencilTexture();
+        expect(texture).toBeDefined();
+        expect(texture.width).toEqual(1);
+        expect(texture.height).toEqual(1);
+        expect(texture.pixelFormat).toEqual(PixelFormat.DEPTH_STENCIL);
+        expect(texture.pixelDatatype).toEqual(PixelDatatype.UNSIGNED_INT_24_8);
+      } else {
+        var renderbuffer = fbm.getDepthStencilRenderbuffer();
+        expect(renderbuffer).toBeDefined();
+        expect(renderbuffer.width).toEqual(1);
+        expect(renderbuffer.height).toEqual(1);
+        expect(renderbuffer.format).toEqual(RenderbufferFormat.DEPTH_STENCIL);
+      }
+    });
+
+    it("creates renderbuffer depth attachments if supportsDepthTexture is false", function () {
+      fbm = new FramebufferManager({
+        depth: true,
+      });
+      fbm.update(context, 1, 1);
+      expect(fbm.getDepthTexture()).toBeUndefined();
+      expect(fbm.getDepthRenderbuffer()).toBeDefined();
+    });
+
+    it("creates renderbuffer depth attachments if supportsDepthTexture is true but context.depthTexture is false", function () {
+      fbm = new FramebufferManager({
+        depth: true,
+        supportsDepthTexture: true,
+      });
+      context._depthTexture = false;
```

Comment:
`_depthTexture` should be set back to its original state so as not to interfere with subsequent tests. See PickingSpec for an example https://github.com/CesiumGS/cesium/blob/4c29dd20873c1bbbb09387bf562f45f24ecc964b/Specs/Scene/PickingSpec.js#L1292-L1308